magpie/api/management/network_node/network_node_views.py

```diff
@@ -0,0 +1,89 @@
+from pyramid.httpexceptions import (
+    HTTPBadRequest,
+    HTTPConflict,
+    HTTPNotFound,
+    HTTPOk,
+    HTTPCreated,
+    HTTPInternalServerError
+)
+
+from pyramid.view import view_config
+
+from magpie import models
+from magpie.api import exception as ax
+from magpie.api import requests as ar
+from magpie.api import schemas as s
+from magpie.api.management.network_node.network_node_utils import create_network_node, delete_network_node
+from magpie.constants import get_constant
+
+
+if get_constant("MAGPIE_NETWORK_MODE", settings_name="magpie.network_mode"):
+    @s.NetworkNodesAPI.get(tags=[s.NetworkNodeTag], response_schemas=s.NetworkNodes_GET_responses)
+    @view_config(route_name=s.NetworkNodesAPI.name, request_method="GET")
+    def get_network_nodes_view(request):
+        nodes = [{"name": n.name, "url": n.url} for n in request.db.query(models.NetworkNode).all()]
+        return ax.valid_http(http_success=HTTPOk, detail=s.NetworkNode_GET_OkResponseSchema.description,
+                             content={"nodes": nodes})
+
+
+    @s.NetworkNodeAPI.get(tags=[s.NetworkNodeTag], response_schemas=s.NetworkNode_GET_responses)
+    @view_config(route_name=s.NetworkNodeAPI.name, request_method="GET")
+    def get_network_node_view(request):
+        node_name = ar.get_value_matchdict_checked(request, "node")
+        node = ax.evaluate_call(
+            lambda: request.db.query(models.NetworkNode).filter(models.NetworkNode.name == node_name).one(),
+            http_error=HTTPNotFound,
+            msg_on_fail=s.NetworkNode_GET_NotFoundResponseSchema.description)
+        return ax.valid_http(http_success=HTTPOk, detail=s.NetworkNode_GET_OkResponseSchema.description,
+                             content={"name": node.name, "url": node.url})
+
+
+    @s.NetworkNodesAPI.post(schema=s.NetworkNode_POST_RequestBodySchema, tags=[s.NetworkNodeTag],
+                            response_schemas=s.NetworkNodes_POST_responses)
+    @view_config(route_name=s.NetworkNodesAPI.name, request_method="POST")
+    def create_network_node_view(request):
+        node_name = request.POST.get("name")
+        node_url = request.POST.get("url")
+        ax.verify_param(all([node_name, node_url]), is_true=True,
```

Comment:
Should use the checking utilities to validate each parameter against some regex. This only ensures there is a non-empty string, but it could be any value that could cause issues if they are malformed.